lib/publiccloud/azure.pm

```diff
@@ -124,16 +131,61 @@ sub upload_img {
 
     my $key = $self->get_storage_account_keys($storage_account);
 
-    assert_script_run('az storage blob upload --max-connections 4 --account-name '
-          . $storage_account . ' --account-key ' . $key . ' --container-name ' . $self->container
-          . ' --type page --file ' . $file . ' --name ' . $img_name, timeout => 60 * 60 * 2);
-    assert_script_run('az disk create --resource-group ' . $self->resource_group . ' --name ' . $disk_name
-          . ' --source https://' . $storage_account . '.blob.core.windows.net/' . $self->container . '/' . $img_name
-          . ' --hyper-v-generation=' . $sku . ' --architecture=' . $arch);
 
-    assert_script_run('az image create --resource-group ' . $self->resource_group . ' --name ' . $img_name
-          . ' --os-type Linux --hyper-v-generation=' . $sku . ' --source=' . $disk_name);
 
+    # Check if blob already exists
+    my $container = $self->container;
+    my $blobs = script_output("az storage blob list --container-name '$container' --account-name '$storage_account' | jq '.[].name'");
+    $blobs =~ s/^\s+|\s+$//g;    # trim
+    record_info("blobs", $blobs);
+    my @blobs = split(/\n/, $blobs);
+    if (grep(/$img_name/, @blobs)) {
+        record_info('blob', "Blob already exists, omitting upload");
+    } else {
+        # Upload blob to Azure. Keep in mind that VM images need to be a page blob type
+        assert_script_run('az storage blob upload --max-connections 4 --account-name '
+              . $storage_account . ' --account-key ' . $key . ' --container-name ' . $self->container
+              . ' --type page --file ' . $file . ' --name ' . $img_name, timeout => 60 * 60 * 2);
+    }
+
+    if ($arch eq 'Arm64') {
+        # For Arm64 images we need to use the image galleries
+        my $publisher = get_var("PUBLIC_CLOUD_AZURE_PUBLISHER", "qe-c");
+        my $offer = get_required_var("PUBLIC_CLOUD_AZURE_OFFER");
+        my $definition = get_required_var('DISTRI') . '-' . get_required_var('FLAVOR') . '-' . get_required_var('VERSION');
+        $definition = get_var("PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION", uc($definition));
+        $sku = get_required_var("PUBLIC_CLOUD_AZURE_SKU");
+        my $target_regions = get_var("PUBLIC_CLOUD_REGION", "westeurope");
+        $target_regions =~ s/,/ /;    # CLI expects spaces as separation, not commas
+        my $hyperv = $sku =~ 'gen1' ? 'V1' : 'V2';
+        my $subscription = get_subscription();
+        my $sa_url = "/subscriptions/$subscription/resourceGroups/imageGroups/providers/Microsoft.Storage/storageAccounts/$storage_account";
+        my $version = get_required_var('PUBLIC_CLOUD_BUILD');
```

Comment:
this value is very common , what will happen when such value already exists ? are we ok with this behavior ? maybe better use date here ? or at least combination of KIWI build together with this value

Reply:
Azure expects the version to be a three digit value `1.2.3`, so it would be difficult to implement it in any other way.

Reply:
What about this?
```suggestion
        my $build = get_required_var('PUBLIC_CLOUD_BUILD');
```

Reply:
In this context we are using the build number as image gallery version - See line 179:

    az sig image-version create ... --gallery-image-version '$version ...

I think `$version` is the better variable name, because it underlines the usage/intention and not the origin.

Reply:
ok , than "DD.MM.YY" ?

Reply:
We can do that, but then we cannot have multiple builds on the same day, which is arguably equally bad. However we could use the digits of the kiwi build number as the third version number in here, would that make more sense to you?

Reply:
currently every KIWI build reset count of PUBLIC_CLOUD_BUILD , so : 
1.  PUBLIC_CLOUD_KIWI_BUILD=0.9.0 PUBLIC_CLOUD_BUILD= 1.00
2.  PUBLIC_CLOUD_KIWI_BUILD=0.9.0 PUBLIC_CLOUD_BUILD= 1.01
3.  PUBLIC_CLOUD_KIWI_BUILD=0.9.0 PUBLIC_CLOUD_BUILD= 1.02
4.  PUBLIC_CLOUD_KIWI_BUILD=0.9.1 PUBLIC_CLOUD_BUILD= 1.00
5.  PUBLIC_CLOUD_KIWI_BUILD=0.9.1 PUBLIC_CLOUD_BUILD= 1.01

so while my idea will make several images during one day overwrite each other , current approach also have some scenarios when it may happen .

Reply:
Done by introducing the `calc_img_version` subroutine.